extensions/resteasy-reactive/quarkus-resteasy-reactive/runtime/src/main/java/io/quarkus/resteasy/reactive/server/runtime/MultipartFormHandler.java

```diff
@@ -0,0 +1,221 @@
+        // it's possible we've already read the entity
+        if (context.hasInputStream()) {
+            // let's not set it twice
+            return;
+        }
+        if (context.serverRequest().getRequestMethod().equals(HttpMethod.GET) ||
+                context.serverRequest().getRequestMethod().equals(HttpMethod.HEAD)) {
+            return;
+        }
+        HttpServerRequest httpServerRequest = context.serverRequest().unwrap(HttpServerRequest.class);
+        if (httpServerRequest.isEnded()) {
+            context.setInputStream(NO_BYTES_INPUT_STREAM);
+        } else {
+            httpServerRequest.setExpectMultipart(true);
+            httpServerRequest.pause();
+            context.suspend();
+            MultipartFormVertxHandler handler = new MultipartFormVertxHandler(context);
+            httpServerRequest.handler(handler);
+            httpServerRequest.endHandler(new Handler<Void>() {
+                @Override
+                public void handle(Void event) {
+                    handler.end();
+                }
+            });
+            httpServerRequest.resume();
+        }
+    }
+
+    private class MultipartFormVertxHandler implements Handler<Buffer> {
+        private final ResteasyReactiveRequestContext rrContext;
+        private final RoutingContext context;
+
+        boolean failed;
+        AtomicInteger uploadCount = new AtomicInteger();
+        AtomicBoolean cleanup = new AtomicBoolean(false);
+        boolean ended;
+        long uploadSize = 0L;
+
+        public MultipartFormVertxHandler(ResteasyReactiveRequestContext rrContext) {
+            this.rrContext = rrContext;
+            this.context = rrContext.serverRequest().unwrap(RoutingContext.class);
+            Set<FileUpload> fileUploads = context.fileUploads();
+
+            context.request().setExpectMultipart(true);
+            if (handleFileUploads) {
+                makeUploadDir(context.vertx().fileSystem());
+            }
+            context.request().uploadHandler(new Handler<HttpServerFileUpload>() {
+                @Override
+                public void handle(HttpServerFileUpload upload) {
+                    if (maxBodySize != -1 && upload.isSizeAvailable()) {
+                        // we can try to abort even before the upload starts
+                        long size = uploadSize + upload.size();
+                        if (size > maxBodySize) {
+                            failed = true;
+                            rrContext.resume(new WebApplicationException(Response.Status.REQUEST_ENTITY_TOO_LARGE));
+                        }
+                    }
+                    if (handleFileUploads) {
+                        // we actually upload to a file with a generated filename
+                        uploadCount.incrementAndGet();
+                        String uploadedFileName = new File(uploadsDirectory, UUID.randomUUID().toString()).getPath();
```

Comment:
This will create a file for every form field, so e.g. if you have a form with a checkbox and a file upload the results of the checkbox will be written out to disk as a file. I think we need to be a bit more selective here, and only write out parts that have a filename set.

Reply:
Actually that's not the case, as Vert.x already handles this under the hood.
The `uploadHandler` is only invoked for parts of the HTTP body that contain `filename`.
The rest are simply added to `request.formAttributes()`.

I'll update the test case to make that clear